sunkit_image/granule.py

```diff
@@ -0,0 +1,391 @@
+
+        plt.tight_layout()
+
+        # rescale axis
+        l0, b0, w0, h0 = ax0.get_position().bounds
+        newpos = [l0, b0-0.01, w0, h0]
+        ax0.set_position(newpos)
+        l1, b1, w1, h1 = ax1.get_position().bounds
+        newpos = [l1, b0-0.01, w0, h0]
+        ax1.set_position(newpos)
+        l2, b2, w2, h2 = ax2.get_position().bounds
+        newpos = [l0, b2, w0, h0]
+        ax2.set_position(newpos)
+        l3, b3, w3, h3 = ax3.get_position().bounds
+        newpos = [l3, b3, w0, h0]
+        ax3.set_position(newpos)
+
+        # add color bar at top
+        outline = mpe.withStroke(linewidth=5, foreground='black')
+        legax = plt.axes([0.1, 0.1, 0.8, 0.85], alpha=0)
+        legax.axis('off')
+        if mark_dim_centers:
+            labels = ['Granule', 'Intergranule', 'Faculae', 'Dim Centers']
+            custom_lines = [lines.Line2D([0], [0], color='white', lw=4,
+                                         path_effects=[outline]),
+                            lines.Line2D([0], [0], color='black', lw=4),
+                            lines.Line2D([0], [0], color="#ffc406", lw=4),
+                            lines.Line2D([0], [0], color='blue', lw=4)]
+            ncol = 4
+        if not mark_dim_centers:
+            labels = ['Granule', 'Intergranule', 'Faculae']
+            custom_lines = [lines.Line2D([0], [0], color='white', lw=4,
+                                         path_effects=[outline]),
+                            lines.Line2D([0], [0], color='black', lw=4),
+                            lines.Line2D([0], [0], color="#ffc406", lw=4)]
+            ncol = 3
+        legax.legend(custom_lines, labels, loc='upper center', ncol=ncol,
+                     fontsize='x-large')
+
+        if not os.path.exists(out_dir):
+            try:
+                os.mkdir(out_dir)
+            except Exception:
+                raise OSError('Could not make directory ' + out_dir)
+
+        plt.savefig(out_dir + 'segmentation_plots_' + id + '.png')
+
+    # convert segmentated image back into SunPy map with original header
+    segmented_map = sunpy.map.Map(segmented_image_markfac, header)
+
+    return segmented_map
+
+def get_threshold(data, method):
+    """
+    Get the threshold value using given skimage segmentation type.
+    ----------
+    Parameters:
+        data (numpy array): data to threshold
+        method (string): skimage thresholding method - options are 'otsu',
+                        'li', 'isodata', 'mean', 'minimum', 'yen', 'triangle'
+    ----------
+    Returns:
+        threshold (float): threshold
+    """
+
+    if not type(data) == np.ndarray:
+        raise ValueError('Input data must be an array.')
+
+    methods = ['otsu', 'li', 'isodata', 'mean', 'minimum', 'yen', 'triangle']
+    if method not in methods:
+        raise ValueError('Method must be one of: ' + str(methods))
+    if method == 'otsu':  # works ok, but classifies low-flux ganules as IG

```

Comment:
These comments seem like useful info, they should be in the docstring.

Reply:
Good point